bundles/org.eclipse.passage.lic.oshi/src/org/eclipse/passage/lic/oshi/State.java

```diff
@@ -96,62 +99,62 @@ List<Swath<?>> swaths() {
 	private void read() throws LicensingException {
 		try {
 			SystemInfo system = new SystemInfo();
-			readOS(system.getOperatingSystem());
-			readHal(system.getHardware());
+			readOS(system::getOperatingSystem);
```

Comment:
`SystemInfo` is an OSHI library class and it's proven to be extremely delicate. We officially do not know how these `get` methods are implemented and we must not rely on any knowing, even if we somehow do know. 

So we cannot rely on whether they cache something or not and thus cannot afford multiple calls. 

We also need to protect ourselves from possible implementation change that comes with version update of the OSHI lib, which is going to happen. 

In any case actual reading must actively happen, so there is no need to switch to suppliers.